app.rb

```diff
@@ -0,0 +1,177 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    i = 0
+    new_array = self
+    new_array = to_a if is_a?(Hash)
+    if is_a?(Range)
+      each do |n|
+        yield n
+      end
+      return self
+    else
+      while i < new_array.length
+        yield new_array[i]
+        i += 1
+      end
+    end
+
+    new_array
+  end
+
+  def my_each_with_index
+    return enum_for unless block_given?
+
+    i = 0
+    j = 1
+    new_array = self
+    new_array = to_a if is_a?(Hash)
+    if is_a?(Range)
+      step(j).each do |n|
+        yield n, j
+        j += 1
+      end
+      return self
+    else
+      while i < new_array.length
+        yield new_array[i], i
+        i += 1
+      end
+    end
+
+    new_array
+  end
```

Comment:
# my_each_with_index

1. [ ]  When a block is given when self is a range behaves like #each_with_index calls the given block once for each element in self
```
Variables:
let(:enum) { array }
```
### Test 👉 `my_each_with_index_output`

|Expected|Got|
|:---:|:---:|
|`"Num: 5, idx: 0\nNum: 6, idx: 1\nNum: 7, idx: 2\nNum: 8, idx: 3\nNum: 9, idx: 4\nNum: 10, idx: 5\nNum...idx: 40\nNum: 46, idx: 41\nNum: 47, idx: 42\nNum: 48, idx: 43\nNum: 49, idx: 44\nNum: 50, idx: 45\n"`|` "Num: 5, idx: 1\nNum: 6, idx: 2\nNum: 7, idx: 3\nNum: 8, idx: 4\nNum: 9, idx: 5\nNum: 10, idx: 6\nNum...idx: 41\nNum: 46, idx: 42\nNum: 47, idx: 43\nNum: 48, idx: 44\nNum: 49, idx: 45\nNum: 50, idx: 46\n"`|